lib/rules/annotation-no-unknown/index.js

```diff
@@ -0,0 +1,88 @@
+'use strict';
+
+const valueParser = require('postcss-value-parser');
+const getDeclarationValue = require('../../utils/getDeclarationValue');
+
+const keywordSets = require('../../reference/keywordSets');
+const optionsMatches = require('../../utils/optionsMatches');
+const report = require('../../utils/report');
+const ruleMessages = require('../../utils/ruleMessages');
+const validateOptions = require('../../utils/validateOptions');
+const { isRegExp, isString } = require('../../utils/validateTypes');
+
+const ruleName = 'annotation-no-unknown';
+
+const messages = ruleMessages(ruleName, {
+	rejected: (annotation) => `Unexpected unknown annotation "${annotation}"`,
+});
+
+const meta = {
+	url: 'https://stylelint.io/user-guide/rules/list/annotation-no-unknown',
+};
+
+/** @type {import('stylelint').Rule} */
+const rule = (primary, secondaryOptions) => {
+	return (root, result) => {
+		const validOptions = validateOptions(
+			result,
+			ruleName,
+			{ actual: primary },
+			{
+				actual: secondaryOptions,
+				possible: {
+					ignoreAnnotations: [isString, isRegExp],
+				},
+				optional: true,
+			},
+		);
+
+		if (!validOptions) {
+			return;
+		}
+
+		root.walkDecls(checkStatement);
+
+		/**
+		 * @param {import('postcss').Declaration} decl
+		 */
+		function checkStatement(decl) {
```

Comment:
Interestingly: if I place a `debugger;` directly under this line, the `value`/`raws` of `decl` never contains `!important` across all the test cases. Seems like I'm doing something wrong!

Reply:
Sorry, I misunderstood. The PostCSS parser assigns the `important` property of a `Declaration` node and omit `!important` from its `value` property if the declaration value includes `!important`. See the example below:

```console
$ node -e 'require("postcss").parse("a { color: red !important }").walkDecls(decl => console.log(decl))'
<ref *1> Declaration {
  raws: { before: ' ', between: ': ' },
  type: 'decl',
  parent: Rule {
    raws: { before: '', between: ' ', semicolon: false, after: ' ' },
    type: 'rule',
    nodes: [ [Circular *1] ],
    parent: Root {
      raws: [Object],
      type: 'root',
      nodes: [Array],
      source: [Object],
      lastEach: 1,
      indexes: [Object],
      [Symbol(isClean)]: false,
      [Symbol(my)]: true
    },
    source: { start: [Object], input: [Input], end: [Object] },
    selector: 'a',
    lastEach: 1,
    indexes: { '1': 0 },
    [Symbol(isClean)]: false,
    [Symbol(my)]: true
  },
  source: {
    start: { offset: 4, line: 1, column: 5 },
    input: Input {
      css: 'a { color: red !important }',
      hasBOM: false,
      id: '<input css 0Is-DD>',
      [Symbol(fromOffsetCache)]: [Array]
    },
    end: { offset: 24, line: 1, column: 25 }
  },
  prop: 'color',
  important: true,
  value: 'red',
  [Symbol(isClean)]: false,
  [Symbol(my)]: true
}
```

So, we may need to treat `!important specially. See also the PostCSS code:

[specially](https://github.com/postcss/postcss/blob/04877f35f481df17737147351b4122944de824f4/lib/parser.js#L259)